### PR TITLE
V2 linter API support

### LINKED ISSUE #126: Support v2 Linter API
In the [v2.3.0 release](https://github.com/steelbrain/linter/releases/tag/v2.3.0) of `linter` support for the v1 API was removed, breaking semver. This means that this provider is no longer supported by the current version of `linter`.

To get this provider working again it will need to be upgrade to the v2 API, a guide for accomplishing that can be found here: [Upgrading to Standard Linter v2](https://github.com/steelbrain/linter/blob/v2.3.0/docs/guides/upgrading-to-standard-linter-v2.md).

Until this is fixed in order to use this provider you will need to downgrade to the last version of `linter` that supports the v1 API, you can do this with `apm install linter@2.2.0`.

### PR DESCRIPTION
**Pull request by vzamanillo**
Fixes #126

### CODE REVIEW

File: lib/init.coffee [hunk 1]
```diff
@@ -85,9 +85,10 @@ module.exports =
 
     @_log 'providing linter, examining javac-callability.'
 
+    name: 'linter-javac'
```
Comment (Arcanemagus) [inline lib/init.coffee]: Usually the name provided here is the name of the source program, `javac` in this case.

File: spec/iss36-spec.coffee [hunk 2]
```diff
@@ -34,10 +34,10 @@ describe 'linter-javac', ->
     it 'should detect all errors and warnings', ->
       expect(@linterBase.parse(@linter_output, @textEditor).length).toBe(4)
 
-    it 'should translate the first message-type to error', ->
+    it 'should translate the first message-severity to error', ->
       firstMessage = @linterBase.parse(@linter_output, @textEditor)[0]
-      expect(firstMessage.type).toMatch('error')
+      expect(firstMessage.severity).toMatch('error')
```
Comment (Arcanemagus) [inline spec/iss36-spec.coffee]: They **must** be exact matches, so this should be `toBe` instead of `toMatch`.

File: spec/iss36-spec.coffee [hunk 3]
```diff
@@ -34,10 +34,10 @@ describe 'linter-javac', ->
...
 
-    it 'should translate the last message-type to warning', ->
+    it 'should translate the last message-severity to warning', ->
       firstMessage = @linterBase.parse(@linter_output, @textEditor)[3]
-      expect(firstMessage.type).toMatch('warning')
+      expect(firstMessage.severity).toMatch('warning')
```
Comment (Arcanemagus) [inline spec/iss36-spec.coffee]: Again, `toBe` instead of `toMatch`.

File: spec/iss36-spec.coffee (hunk 2 above)
Comment (vzamanillo) [reply spec/iss36-spec.coffee]: ouch..

**Pull request merged**